.github/workflows/ci.yml

```diff
@@ -46,13 +46,37 @@ jobs:
         distribution: 'temurin'
         cache: maven
     - name: Test
+      if: github.event_name == 'push' && github.ref_name == 'main'
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
       run: |-
-        mvn -B --no-transfer-progress -pl '!e2e' clean verify \
-          failsafe:integration-test -DskipITs=false \
-          org.sonarsource.scanner.maven:sonar-maven-plugin:sonar
+        mvn -pl '!e2e' clean verify \
+          failsafe:integration-test -DskipITs=false
+    # Publishing coverage to Codacy is only possible for builds of push events.
+    # PRs from forks do not get access to repository secrets.
+    # https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+    - name: Publish test coverage
+      if: ${{ github.event_name != 'pull_request' && github.repository_owner == 'DependencyTrack' }}
+      uses: codacy/codacy-coverage-reporter-action@v1.3.0
+      with:
+        project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+        language: Java
+        coverage-reports: target/jacoco-ut/jacoco.xml
+    - name: Save PR details
+      if: ${{ github.event_name == 'pull_request' }}
+      run: |-
+        echo ${{ github.sha }} > pr-commit.txt
+        echo ${{ github.event.number }} > PR_NUMBER.txt
+    - name: Upload PR test coverage report
+      if: ${{ github.event_name == 'pull_request' }}
+      uses: actions/upload-artifact@v4
+      with:
+        name: pr-test-coverage-report
+        path: |-
+          pr-commit.txt
+          PR_NUMBER.txt
+          target/jacoco-ut/jacoco.xml
```

Comment:
Hyades has multiple modules and as such will produce multiple JaCoCo reports.

You can use [wildcard patterns](https://github.com/actions/upload-artifact?tab=readme-ov-file#upload-using-a-wildcard-pattern) to cover them all.

For reference, after running `mvn clean verify`:

```shell
$ ls -la **/target/site/jacoco/jacoco.xml
-rw-r--r--  1 nscur0  staff    32714 May 24 11:24 commons-kstreams/target/site/jacoco/jacoco.xml
-rw-r--r--  1 nscur0  staff   157244 May 24 11:25 commons-persistence/target/site/jacoco/jacoco.xml
-rw-r--r--  1 nscur0  staff   121177 May 24 11:24 commons/target/site/jacoco/jacoco.xml
-rw-r--r--  1 nscur0  staff   143897 May 24 11:26 mirror-service/target/site/jacoco/jacoco.xml
-rw-r--r--  1 nscur0  staff    99536 May 24 11:27 notification-publisher/target/site/jacoco/jacoco.xml
-rw-r--r--  1 nscur0  staff  4571540 May 24 11:25 proto/target/site/jacoco/jacoco.xml
-rw-r--r--  1 nscur0  staff     2506 May 24 11:24 quarkus-extensions/config-dependencytrack/deployment/target/site/jacoco/jacoco.xml
-rw-r--r--  1 nscur0  staff   111665 May 24 11:27 repository-meta-analyzer/target/site/jacoco/jacoco.xml
-rw-r--r--  1 nscur0  staff   184772 May 24 11:29 vulnerability-analyzer/target/site/jacoco/jacoco.xml
```